frontend/src/components/ViewTrips/add-trip.js

```diff
@@ -0,0 +1,143 @@
+        {formControl}
+      {/* Temporary instructions until fix Issue #52 */}
+      <Form.Text className="text-muted">
+        {subFormText}
+      </Form.Text>
+    </Form.Group>
+  )
+}
+
+/**
+ * Component corresponding to add trips modal.
+ *
+ * @param {Object} props These are the props for this component:
+ * - show: Boolean that determines if the add trips modal should be displayed.
+ * - handleClose: The function that handles closing the add trips modal.
+ * - userEmail: The current users email.
+ *
+ * @extends React.Component
+ */
+class AddTrip extends React.Component {
+  /** @inheritdoc */
+  constructor(props) {
+    super(props);
+
+    this.nameRef = React.createRef();
+    this.descriptionRef = React.createRef();
+    this.destinationRef = React.createRef();
+    this.startDateRef = React.createRef();
+    this.endDateRef = React.createRef();
+    this.collaboratorsRef = React.createRef();
+  }
+
+  /**
+   * Upon submission of the form, a new Trip document is created and the add
+   * trip modal is closed.
+   */
+  submitNewTrip = () => {
+    console.log('New trip submitted.');
+    createTrip({name: this.nameRef.current.value,
+               description: this.descriptionRef.current.value,
+               destination: this.destinationRef.current.value,
+               startDate: this.startDateRef.current.value,
+               endDate: this.endDateRef.current.value,
+               collaborators: this.collaboratorsRef.current.value
+    });
+
+    this.props.handleClose();
+  }
+
+  /** @inheritdoc */
+  render() {
+    return (
+      <Modal show={this.props.show} onHide={this.props.handleClose}>
+        <Modal.Header closeButton>
+          <Modal.Title>Add New Trip</Modal.Title>
+        </Modal.Header>
+
+        <Form>
+          <Modal.Body>
+            {createFormGroup('tripNameGroup', 'Trip Name', 'text',
+                             'Enter Trip Name', this.nameRef)}
+            {createFormGroup('tripDescripGroup', 'Trip Description', 'text',
+                             'Enter Trip Description', this.descriptionRef)}
+            {createFormGroup('tripDestGroup', 'Trip Destination', 'text',
+                             'Enter Trip Destination', this.destinationRef)}
+            {createFormGroup('tripStartDateGroup', 'Start Date', 'text',
+                            'Enter Trip Start Date', this.startDateRef,
+                            'Enter date in the form: \'mm/dd/yyy\'')}
+            {createFormGroup('tripEndDateGroup', 'End Date', 'text',
+                          'Enter Trip End Date', this.endDateRef,
+                          'Enter date in the form: \'mm/dd/yyy\'')}
+            {createFormGroup('tripCollabsGroup', 'Trip Collaborators', 'text',
+                           'Enter Collaborator Emails', this.collaboratorsRef,
+                           'Enter emails in the form: \'user1@email.com, ...,' +
+                           ' userx@email.com\'')}
```

Comment:
Pretty nice how this was abstracted out!

Reply:
Thanks!